src/audio/audio_sink.cpp

```diff
@@ -11,71 +11,146 @@
 #include <cassert>
 #include <climits>
 #include <cstring>
+#include <memory>
 #include <string>
 
-#include "portaudiocpp/PortAudioCpp.hxx"
+#include "SDL.h"
 
 #include "../errors.hpp"
-#include "../sample_formats.hpp"
 #include "../messages.h"
 #include "audio_sink.hpp"
 #include "audio_source.hpp"
 #include "ringbuffer.hpp"
+#include "sample_formats.hpp"
 
-const size_t AudioSink::RINGBUF_POWER = 16;
+const size_t SdlAudioSink::RINGBUF_POWER = 16;
 
-AudioSink::AudioSink(const AudioSource &source,
-                     const AudioSinkConfigurator &conf)
+/**
+ * The callback used by SDL_Audio.
+ * Trampolines back into vsink, which must point to an SdlAudioSink.
+ */
+static void SDLCallback(void *vsink, std::uint8_t *data, int len)
+{
+	assert(vsink != nullptr);
+	auto sink = static_cast<SdlAudioSink *>(vsink);
+	sink->Callback(data, len);
+}
+
+/* static */ std::unique_ptr<AudioSink> SdlAudioSink::Build(
+                const AudioSource &source, int device_id)
```

Comment:
is this clang format doing this? maybe the weighting needs further adjustment, because ew